Libraries/PermissionsAndroid/PermissionsAndroid.js

```diff
@@ -111,8 +133,26 @@ class PermissionsAndroid {
    * necessary to show a dialog explaining why the permission is needed
    * (https://developer.android.com/training/permissions/requesting.html#explain)
    * and then shows the system permission dialog
+   *
+   * @deprecated
    */
   async requestPermission(permission: string, rationale?: Rationale) : Promise<boolean> {
+    console.warn('"PermissionsAndroid.requestPermission" is deprecated. Use "PermissionsAndroid.request" instead');
+    const response = await this.request(permission, rationale);
+    return (response === this.RESULTS.GRANTED);
+  }
+
+  /**
+   * Prompts the user to enable a permission and returns a promise resolving to a
+   * string value indicating whether the user allowed or denied the request
+   *
+   * If the optional rationale argument is included (which is an object with a
+   * `title` and `message`), this function checks with the OS whether it is
+   * necessary to show a dialog explaining why the permission is needed
+   * (https://developer.android.com/training/permissions/requesting.html#explain)
+   * and then shows the system permission dialog
+   */
+  async request(permission: string, rationale?: Rationale) : Promise<string> {
```

Comment:
Add `type PermissionStatus = 'granted' | 'denied' | 'never_ask_again'` at the top and change `Promise<string>` to `Promise<PermissionStatus>`